crates/lsp/src/capabilities.rs

```diff
@@ -0,0 +1,40 @@
+use tower_lsp::lsp_types::ClientCapabilities;
+use tower_lsp::lsp_types::PositionEncodingKind;
+
+/// A resolved representation of the [ClientCapabilities] the Client sends over that we
+/// actually do something with
+#[derive(Debug, Default)]
+pub(crate) struct ResolvedClientCapabilities {
+    pub(crate) position_encodings: Vec<PositionEncodingKind>,
+    pub(crate) dynamic_registration_for_did_change_configuration: bool,
+    pub(crate) dynamic_registration_for_did_change_watched_files: bool,
+}
```

Comment:
Collecting client capabilities we care about here. We get them in `Initialize` but use them in other places